front/src/modules/timesStops/TimesStopsOutput.tsx

```diff
@@ -0,0 +1,235 @@
+      trainSimulation.base,
+      opPoint?.position * 1000,
+      'times'
+    );
+    const nextOpPointBaseTime = interpolateValue(
+      trainSimulation.base,
+      nextOpPoint?.position * 1000,
+      'times'
+    );
+    const baseDuration = nextOpPointBaseTime - opPointBaseTime;
+    const opPointprovTime = interpolateValue(
+      trainSimulation.provisional,
+      opPoint?.position * 1000,
+      'times'
+    );
+    const nextOpPointProvTime = interpolateValue(
+      trainSimulation.provisional,
+      nextOpPoint?.position * 1000,
+      'times'
+    );
+    const provisionalDuration = nextOpPointProvTime - opPointprovTime;
+    // Time between current scheduled point and next scheduled point
+    const realLostTime = provisionalDuration - baseDuration;
+    const distanceToNextOpPoint = nextOpPoint?.position - opPoint?.position;
+    let realMargin = 0;
+    let diffMargins = 0;
+    if (marginUnit === '%') {
+      realMargin = 100 * (realLostTime / baseDuration) || 0;
+      if (theoreticalMargin) {
+        const theoreticalMarginValue = Number(theoMarginRegexMatches?.[1]);
+        const theoriticalLostTime = baseDuration * (theoreticalMarginValue * 0.01);
+        diffMargins = (realLostTime - theoriticalLostTime) * 0.001;
+      }
+    } else {
+      // unit is min/100km
+      realMargin = (100 * (realLostTime / 60)) / distanceToNextOpPoint || 0;
+      if (theoreticalMargin) {
+        const theoreticalMarginValue = Number(theoMarginRegexMatches?.[1]);
+        const theoriticalLostTime = (60 * theoreticalMarginValue * distanceToNextOpPoint) / 100;
+        diffMargins = (realLostTime - theoriticalLostTime) * 0.001;
+      }
+    }
+
+    const theoreticalMarginWithUnit = formatDigitsAndUnit(theoreticalMargin);
+    const realMarginWithUnit = formatDigitsAndUnit(String(realMargin), marginUnit);
+    const diffMarginsWithUnit = diffMargins ? formatDigitsAndUnit(String(diffMargins), 's') : '';
+    return [theoreticalMarginWithUnit, realMarginWithUnit, diffMarginsWithUnit];
+  }
+  return ['', '', ''];
+};
+
+const TimesStopsOutput = ({
+  train,
+  simulatedTrain,
+  pathProperties, // raw from simulation result, calculated values
+  infraId,
+}: TimesStopsOutputProps) => {
+  const simResults = useSimulationResults();
+  usePathfindingV2(); // is it really needed to add position_on_path to pathSteps ? use position_on_path to know where the via is in the suggested point array
+  const { getPathSteps, getStartTime } = useOsrdConfSelectors();
+  const pathSteps = useSelector(getPathSteps);
+  const startTime = useSelector(getStartTime);
+  const selectedTrainId = useSelector(getSelectedTrainId);
+
+  const { data: selectedTrainSchedule = {} as TrainScheduleResult } =
+    osrdEditoastApi.endpoints.getV2TrainScheduleById.useQuery(
+      {
+        id: selectedTrainId as number,
+      },
+      { skip: !selectedTrainId }
+    );
+
+  const tsScheduleByAt = keyBy(selectedTrainSchedule.schedule, 'at');
```

Comment:
everywhere you create a const that starts with `ts`, can't you juste remove this start? I think `scheduleByAt` or `pathPointsWithOpPointIndices` or even `pathByUic`.. is enough

Reply:
it was too indicate that it’s data derived from train schedule. But yeah not really necessary.
It’s tough to find good names